hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/ITestAzureBlobFileSystemLease.java

```diff
@@ -362,17 +370,39 @@ public void testAcquireRetry() throws Exception {
         .doCallRealMethod().when(mockClient)
         .acquireLease(anyString(), anyInt(), any(TracingContext.class));
 
-    lease = new AbfsLease(mockClient, testFilePath.toUri().getPath(), 5, 1, tracingContext);
+    doThrow(new AbfsLease.LeaseException("failed to acquire 1"))
+        .doThrow(new AbfsLease.LeaseException("failed to acquire 2"))
+        .doCallRealMethod().when(mockClient)
+        .acquireBlobLease(anyString(), anyInt(), any(TracingContext.class));
+
+    if (getPrefixMode(fs) == PrefixMode.BLOB) {
+      lease = new AbfsBlobLease(mockClient, testFilePath.toUri().getPath(), 5,
+          1, null,
+          tracingContext);
+    } else {
+      lease = new AbfsDfsLease(mockClient, testFilePath.toUri().getPath(), 5, 1,
+          null,
+          tracingContext);
+    }
     Assert.assertNotNull("Acquire lease should have retried", lease.getLeaseID());
     lease.free();
     Assert.assertEquals("Unexpected acquire retry count", 2, lease.getAcquireRetryCount());
 
     doThrow(new AbfsLease.LeaseException("failed to acquire")).when(mockClient)
         .acquireLease(anyString(), anyInt(), any(TracingContext.class));
 
+    doThrow(new AbfsLease.LeaseException("failed to acquire")).when(mockClient)
+        .acquireBlobLease(anyString(), anyInt(), any(TracingContext.class));
+
     LambdaTestUtils.intercept(AzureBlobFileSystemException.class, () -> {
-      new AbfsLease(mockClient, testFilePath.toUri().getPath(), 5, 1,
-          tracingContext);
+      if (getPrefixMode(fs) == PrefixMode.BLOB) {
```

Comment:
common code, can be made into a function call

Reply:
What is common code here. Kindly elaborate please.